### DISCUSSION TITLE
Can't delete a device with more than 110 interfaces

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#12022] [Help Wanted!]
### NetBox version

v3.4.6

### Python version

3.9

### Steps to Reproduce

Choose a device with a lot of interfaces (>110 interfaces) and try to delete it using the UI Delete button.

### Expected Behavior

The deletion process should succeed.

### Observed Behavior

After 30 seconds, the request will get timed out and the following logs will be displayed on the POD logs:
~~~
[2023-03-21 09:59:21 +0000] [12] [DEBUG] POST /dcim/devices/20337/delete/
[2023-03-21 09:59:52 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:12)
[2023-03-21 09:59:52 +0000] [12] [INFO] Worker exiting (pid: 12)
[2023-03-21 09:59:53 +0000] [1] [WARNING] Worker with pid 12 was terminated due to signal 9
[2023-03-21 09:59:53 +0000] [16] [INFO] Booting worker with pid: 16
[2023-03-21 09:59:59 +0000] [16] [DEBUG] Ignoring EPIPE
~~~

This is causing the pod to terminate and no HTTP response will be sent back to the client.

**Comment by jeremystretch** [comment]
Your WSGI worker is timing out before the operation completes. 30 seconds is rather low. You can either configure a longer timeout for your worker, or delete the interfaces in smaller batches prior to deleting the device.

**Reply by ghost** [reply]
Thanks for the answer, it solved the issue which is deleting the target device. However, if I hit delete, then after couple of seconds a timeout message still appear again but this time if I refresh the page after the deletion then the UI will show that the device is deleted.

**Reply by ghost** [reply]
I'v added these options to the gunicorn configs but I still get timeout  after 30s when trying to open some pages with high number of entries (i.e. prefixes).
I checked all the componants (reverseproxy, loadbalancer) and all the timeout values are >=120s.
```
workers = 17
timeout = 120
max_requests = 1000
max_requests_jitter = 200
```

**Reply by kkthxbye-code** [reply]
There's nothing in netbox itself that would cause this, so you will need to go through all your other components again to find the place where a 30 second timeout is still in effect.

**Reply by baldgeek** [reply]
I am assuming you are doing so, but make sure you are restarting services after config changes to load the new timeouts.

**Reply by abhi1693** [reply]
Make sure to look at your ingress configuration as well. It could very well be still at default of 30 seconds

**Reply by ghost** [reply]
Ok I'll see what I can do. But now I guess we can close this discussion.
Thank you all for your replies.
